src/main/java/org/opentripplanner/middleware/otp/OtpRequestProcessor.java

```diff
@@ -13,81 +13,121 @@
+ * is of interest the response is intercepted and processed. In all cases, the response from OTP (content and HTTP status)
+ * is passed back to the requester.
  */
 public class OtpRequestProcessor {
     private static final Logger LOG = LoggerFactory.getLogger(OtpRequestProcessor.class);
     private static final String OTP_SERVER = getConfigPropertyAsText("OTP_SERVER");
-    private static final String OTP_SERVER_PLAN_END_POINT = getConfigPropertyAsText("OTP_SERVER_PLAN_END_POINT");
+    private static final String OTP_ROOT_ENDPOINT = getConfigPropertyAsText("OTP_ROOT_ENDPOINT");
+    private static final String OTP_PLAN_ENDPOINT = getConfigPropertyAsText("OTP_PLAN_ENDPOINT");
 
     /**
-     * Obtain and process plan response from OpenTripPlanner. Store the response if consent is given. Handle the
-     * process and all exceptions seamlessly so as not to affect the response provided to the requester.
+     * Register http endpoint with {@link spark.Spark} instance based on the OTP root endpoint. An OTP root endpoint is
+     * required to distinguish between OTP and other middleware requests.
      */
-    public static String planning(Request request, spark.Response response) {
+    public static void register(Service spark) {
+        // available at (depending on config) http://localhost:4567/otp/*
+        spark.get(OTP_ROOT_ENDPOINT + "/*", OtpRequestProcessor::proxy);
+    }
+
+    /**
+     * Inspect all requests that are made to OTP. If the request is of interest (for instance plan response) intercept
+     * the response from OTP and process accordingly. In all cases, pass the response back to the requester.
+     */
+    private static String proxy(Request request, spark.Response response) {
         if (OTP_SERVER == null) {
             logMessageAndHalt(request, HttpStatus.INTERNAL_SERVER_ERROR_500, "No OTP Server provided, check config.");
             return null;
         }
 
-        String batchId = request.queryParams("batchId");
-        if (batchId == null) {
-            //TODO place holder for now
-            batchId = "-1";
-        }
-
         // attempt to get response from OTP server based on requester's query parameters
-        OtpDispatcher otpDispatcher = new OtpDispatcherImpl(OTP_SERVER);
-        OtpDispatcherResponse otpDispatcherResponse = otpDispatcher.getPlan(request.queryString(), OTP_SERVER_PLAN_END_POINT);
+        OtpDispatcherResponse otpDispatcherResponse = OtpDispatcher.serviceRequest(request.queryString(), request.uri());
         if (otpDispatcherResponse == null || otpDispatcherResponse.responseBody == null) {
             logMessageAndHalt(request, HttpStatus.INTERNAL_SERVER_ERROR_500, "No response from OTP server.");
             return null;
         }
 
+        String targetService = extractTargetService(request.uri());
+        // if the target service is plan, process response
+        if (targetService.equalsIgnoreCase(OTP_PLAN_ENDPOINT)) plan(request, otpDispatcherResponse);
```

Comment:
We may just want to check that the request URI ends with `/plan` as mentioned above.

Reply:
Updated.